dataclasses_avroschema/schema_generator.py

```diff
@@ -67,10 +68,44 @@ def get_fields(cls: typing.Any) -> typing.List[FieldType]:
             return cls.generate_schema().fields
         return cls.schema_def.fields
 
+    @classmethod
+    def _get_enum_type_map(cls: typing.Any) -> typing.Dict[str, enum.Enum]:
+        enum_types = {}
+        for field_type in cls.get_fields():
+            if isinstance(field_type, EnumField):
+                enum_types[field_type.name] = field_type.type
+            elif isinstance(field_type, UnionField):
+                for sub_type in field_type.type.__args__:
+                    if issubclass(sub_type, enum.Enum):
```

Comment:
Amazing job @aaronkub !! I see one issue in `deserialization`:

```python
import datetime
import uuid
import typing

from dataclasses_avroschema import AvroModel, types


class UnionSchema(AvroModel):
    "Some Unions"
    logical_union: typing.Union[datetime.datetime, datetime.date, uuid.uuid4]

instance = UnionSchema.fake()

# GOOD!!
In [67]: instance.serialize()
Out[67]: b'\x04H664f2e44-5d63-46ca-95a9-c4ae6d5d6d70'


In [70]: UnionSchema.deserialize(instance.serialize())
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-70-f795bc1d86fa> in <module>
----> 1 UnionSchema.deserialize(instance.serialize())

~/Projects/dataclasses-schema-generator/dataclasses_avroschema/schema_generator.py in deserialize(cls, data, serialization_type, create_instance, writer_schema)
    140         )
    141
--> 142         output = cls._deserialize_complex_types(payload)
    143
    144         if create_instance:

~/Projects/dataclasses-schema-generator/dataclasses_avroschema/schema_generator.py in _deserialize_complex_types(cls, payload)
     86     def _deserialize_complex_types(cls, payload: typing.Dict[str, typing.Any]) -> typing.Any:
     87         output = {}
---> 88         enum_type_map = cls._get_enum_type_map()
     89         for field, value in payload.items():
     90             if isinstance(value, dict):

~/Projects/dataclasses-schema-generator/dataclasses_avroschema/schema_generator.py in _get_enum_type_map(cls)
     77             elif isinstance(field_type, UnionField):
     78                 for sub_type in field_type.type.__args__:
---> 79                     if issubclass(sub_type, enum.Enum):
     80                         enum_types[field_type.name] = sub_type
     81             elif isinstance(field_type, RecordField):

TypeError: issubclass() arg 1 must be a class
```

Reply:
Thanks @marcosschroh! I will get this resolved ASAP.

Reply:
I've resolved this issue and added in your example as a unit test.